miio/gateway.py

```diff
@@ -536,38 +513,216 @@ class SubDevice:
     def __init__(self, gw, sid, type_, _, __, ___):
         self.gw = gw
         self.sid = sid
-        self.type = DeviceType(type_)
+        self._battery = None
+        try:
+            self.type = DeviceType(type_)
+        except ValueError:
+            self.type = DeviceType(-1)
```

Comment:
The type should be set in the inheriting class' init.

Reply:
Why?
In this way you only need to write the code ones, otherwise this needs to be copied into all device_type specific classes

Reply:
The idea with having this in a class-specific init is a commonly used pattern and initialized class will/should always know its type by its own.

However, this is not necessary anymore at all as the type is defined already at the initialization?

Reply:
bump.